students/Shweta/Lesson05/mailroom_p3.py

```diff
@@ -0,0 +1,113 @@
+        }
+
+'''main message displayed on execution of this job'''
+prompt_msg="\n".join(("Welcome for Donation Program !!",
+                     "Please choose from following:",
+                     "1- Send a Thank you",
+                     "2- Create a Report",
+                     "3- Send a letter to all the donors",
+                     "4- Quit",
+                     "Type a number to select >>> "))
+
+'''message displayed for selecting the donors out of list'''
+mty_prompt= "\n".join(("Please type the full name of the donar OR",
+                       "type 'list' to see a list of donors",
+                       "Type input here -->"))
+
+'''thankyou message for donation'''
+mty_msg="\n".join(("Thankyou {}",
+                   "for your generous donation of {:.2f}",))
+
+
+ltr="\n".join(("", "Dear {},",
+              "Thank you for your very kind donation of ${:.2f}.",
+              "It will be put to very good use.",
+              "           Sincerely,",
+              "               -The Team"))
+
+
+
+###########Thankyou Function##############
+def send_thankyou():
+    '''task 1 of the mail room- send thankyou for donations'''
+    dname=input(mty_prompt)
+    if dname == 'list':
+        create_report()
+    else:
+        try:
+            damt=float(input("Enter your donation amount in dollars --> "))
+        except ValueError:
+            print('\n Not a valid amount, try again')
+        else:
+            if donar_db.get(dname) == None:
+                donar_db[dname]=[damt]
+            else:
+                donor_db.get(dname).append(damt)
+
+        input(mty_msg.format(dname,damt))
+
+
+##########send letter#################
+def send_letter():
+    for dn,da in donar_db.items():
+        with open(dn + ".txt",'w') as infile:
+            infile.write(ltr.format(dn,sum(da)))
+           
+    print("Your letter is in {} ".format(os.getcwd()))
+
+###########Create Report ################
+        
+def create_report():
+    '''task 2 of the mailroom - create report'''       
+    top='{:20}| {:>20}|{:>20}| {:>20}'.format('Donor Name', 'Total Given','Num Gifts', 'Average Gift')
+    print(top)
+    print('-'*100)
+
+    for key,value in donar_db.items():
+        line='{:20} ${:>20.2f}   {:20}   ${:20.2f}'.format(key,sum(value),len(value),(sum(value)/len(value)))
+        print(line)
+    print('')
+
+
+##########Exit Program#################
+
+def quit_pgm():
+    ''' exit the program cleanly'''
+    print("Bye!")
+    sys.exit()
+
+
+###########Main block##################
+def main():
+    '''Main block of the code'''      
+    while True:
+        response = input(prompt_msg)
+        if response == "1":
+            send_thankyou()
+        elif response == "2":
+            create_report()
+        elif response == "3":
+            send_letter()
+        elif response == "4":
+            quit_pgm()
+        else:
+            print("Please enter valid option!")
```

Comment:
the code should advance on previous version of mailrooms and build on top of it - so using switch dict and sorted report are part of that.